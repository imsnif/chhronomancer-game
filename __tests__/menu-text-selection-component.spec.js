import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('MenuTextSelection (selected) => renders correctly', () => {
  const MenuTextSelection = require('../components/menu-text-selection').default
  const tree = renderer.create(
    <MenuTextSelection
      onPress={() => {}}
      text='foo'
      selected
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('MenuTextSelection (unselected) => renders correctly', () => {
  const MenuTextSelection = require('../components/menu-text-selection').default
  const tree = renderer.create(
    <MenuTextSelection
      onPress={() => {}}
      text='foo'
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
