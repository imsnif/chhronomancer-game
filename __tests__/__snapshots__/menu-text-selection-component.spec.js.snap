// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuTextSelection (selected) => renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "flex": 1,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#329932",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "height": 9,
        "includeFontPadding": false,
        "lineHeight": 9,
        "textAlign": "center",
        "textAlignVertical": "bottom",
      }
    }
  >
    [ foo ]
  </Text>
</View>
`;

exports[`MenuTextSelection (unselected) => renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "flex": 1,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#329932",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "height": 9,
        "includeFontPadding": false,
        "lineHeight": 9,
        "textAlign": "center",
        "textAlignVertical": "bottom",
      }
    }
  >
    foo
  </Text>
</View>
`;
